cluster-provision/gocli/pkg/libssh/ssh.go

```diff
@@ -52,23 +62,14 @@ func NewSSHClient(port uint16, idx int, root bool) (*SSHClientImpl, error) {
 // SSH performs two ssh connections, one to the forwarded port by dnsmasq to the local which is the ssh port of the control plane node
 // then a hop to the designated host where the command is desired to be ran
 func (s *SSHClientImpl) Command(cmd string) error {
-	client, err := ssh.Dial("tcp", net.JoinHostPort("127.0.0.1", fmt.Sprint(s.sshPort)), s.config)
-	if err != nil {
-		return fmt.Errorf("Failed to connect to SSH server: %v", err)
-	}
-	defer client.Close()
-
-	conn, err := client.Dial("tcp", fmt.Sprintf("192.168.66.10%d:22", s.nodeIdx))
-	if err != nil {
-		return fmt.Errorf("Error establishing connection to the next hop host: %s", err)
+	if s.client == nil {
+		err := s.initClient()
+		if err != nil {
+			return err
+		}
```

Comment:
Why was this change made? I am not seeing any explanation.

Reply:
this code (the client instantiation) is duplicated around all ssh methods.
both Daniel and Antonio on an earlier review thought its better to reuse the connection. and i agree with this as well. as it avoids the overhead of connection establishment with every command

Reply:
It would be great to make this change explicit by making it a separate commit